Order specs by alphabetical order

ryan-blakley opened the pull request:
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
* Reorder all specs into alphabetical order.
* Additionally I moved the specs in default that are only dep specs used by other specs to the top, since they need to be defined before the spec that uses them.
* Removed the multiple rpm specs that didn't accomplish anything but reference each other.
* Removed a duplicate spec entry in the registry.

xiangce reviewed (approved):
That's great. Thanks for reordering!

I think it can be merged once the conflict is resolved.
A new spec (container_installed_rpms) was added in PR#3560, the PR was just merged and caused the conflict and the `rpm_format` might need to be moved back. Sorry about this inconvenience.

ryan-blakley commented:
> That's great. Thanks for reordering!
> 
> I think it can be merged once the conflict is resolved. A new spec (container_installed_rpms) was added in PR#3560, the PR was just merged and caused the conflict and the `rpm_format` might need to be moved back. Sorry about this inconvenience.

@xiangce Thanks for the heads up, I fixed the conflict and replaced the format_rpm with the _make_rpm_formatter like I did for the installed_rpms spec. If someone who has access to a container host could make sure using _make_rpm_formatter directly works correctly in the container spec it would be great to confirm.
